[Conversation API] initial docs setup (#4419)
dapr/docs: daprdocs/content/en/reference/api/conversation_api.md

---
type: docs
title: "Conversation API reference"
linkTitle: "Conversation API"
description: "Detailed documentation on the conversation API"
weight: 1400
---

> Dapr provides users with an abstraction layer to reduce the complexity of interacting with LLMS using either a new runtime method or HTTP endpoint.

Reviewer: ```suggestion
Dapr provides an API to interact with Large Language Models (LLMs) and enables critical performance and security functionality with features like prompt caching and PII data obfuscation.
```

> Dapr provides users with an abstraction layer to reduce the complexity of interacting with LLMS using either a new runtime method or HTTP endpoint.

Reviewer 2: `Dapr provides users with an abstraction layer to reduce the complexity of interacting with LLMS using either an HTTP endpoint or SDK methods`.

## Converse

This endpoint lets you converse with LLMs.

```

> POST /v1.0/conversation/[component]/converse

Reviewer: ```suggestion
POST /v1.0/conversation/<llm-name>/converse
```

> POST /v1.0/conversation/[component]/converse

Reviewer 2: v1.0-alpha1

```

### URL parameters

| Parameter | Description |
| --------- | ----------- |

> | `component` | The name of the component you're using. [See a list of all available Conversation components]({{< ref supported-conversation >}})

Reviewer: ```suggestion
| `llm-name` | The name of the LLM component. [See a list of all available conversation components]({{< ref supported-conversation >}})
```

> 

Reviewer: Need to add Request Body fields into a table like this https://docs.dapr.io/reference/api/state_api/#request-body

	conversationContext  - The conversation context to provide to the LLM
	prompt  - The conversation prompt to provide to the LLM
	
	The "inputs" name below should be replace with "prompt"

### Request content

```json
REQUEST = {

>   "conversationContext": "ID of an existing chat room",

Reviewer 2: remove `conversationContext`

  "inputs": ["what is Dapr", "Why use Dapr"],
  "parameters": {},
}
```

### HTTP response codes

Code | Description
---- | -----------
`202`  | Accepted
`400`  | Request was malformed
`500`  | Request formatted correctly, error in dapr code or underlying component

### Response content

```json
RESPONSE  = {

>   "conversationContext": "ID of an existing chat room",

Reviewer 2: remove `conversationContext`
